### PR TITLE
Multiple Choice

### LINKED ISSUE #6: Add a flag "multipleCorrect" to questions
A question can be either multiple correct or single correct. 

- Add a flag to indicate this. "multipleCorrect" : Boolean
- if the question is single correct, only allow one `correctAnswers` or send an error.
- Add similar check when submitting a question and quiz.

### PR DESCRIPTION
**Pull request by vibhorgupta-gh**
fixes #6

### CODE REVIEW

File: routes/api/quiz/controller.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ class QuizController extends BaseController {
         model: DB.questions,
         include: {
           model: DB.choices,
-          attributes: ['id', 'title', 'positiveWeight', 'negativeWeight']
+          attributes: ['id', 'title', 'positiveWeight', 'negativeWeight, multipleCorrect']
```
Comment (abhishek97) [inline routes/api/quiz/controller.js]: Why?

File: routes/api/questions/controller.js [hunk 2]
```diff
@@ -97,7 +113,7 @@ class QuestionsController extends BaseController {
     //   }
     // })
 
-    const { score, correctlyAnswered, incorrectlyAnswered } = U.getScore(markedChoices, U.parseIntArray(question.correctAnswers), question.choices)
+    const { score, correctlyAnswered, incorrectlyAnswered } = U.getScore(markedChoices, U.parseIntArray(question.correctAnswers), possibleChoices)
```
Comment (abhishek97) [inline routes/api/questions/controller.js]: why you changed this?

File: routes/api/questions/controller.js [hunk 3]
```diff
@@ -53,7 +65,7 @@ class QuestionsController extends BaseController {
 
   */
   async submitQuestion (req, res) {
-    let { markedChoices } = req.body
+    let { correctAnswers, markedChoices } = req.body
```
Comment (abhishek97) [inline routes/api/questions/controller.js]: why ?

File: routes/api/quiz/controller.js (hunk 1 above)
Comment (vibhorgupta-gh) [reply routes/api/quiz/controller.js]: to extract the multiple correct attribute from question

File: routes/api/questions/controller.js (hunk 2 above)
Comment (vibhorgupta-gh) [reply routes/api/questions/controller.js]: they were both the same stuff, so i figured pass the variable instead. will change

**Comment by VibhorCodecianGupta** [comment]
@abhishek97 done, made requested changes

**Comment by vibhorgupta-gh** [comment]
@abhishek97 any updates?

**Comment by vibhorgupta-gh** [comment]
@abhishek97 what was the issue with this?

**Comment by abhishek97** [comment]
@VibhorCodecianGupta Nothing, both look identical. Just chose the earlier one.

**Comment by vibhorgupta-gh** [comment]
@abhishek97 wasn't this the earlier one? the usable commit came earliest here i guess. the last commit was just line break fixes and error messages!

**Pull request closed without merging**